### ISSUE TITLE
Can we use here templates?

### ISSUE THREAD

**Issue by poul250** [linksplatform/Ranges#44]
https://github.com/linksplatform/Ranges/blob/d7b5ac881f7e8921520f5264ce04a1d97dfd31c3/cpp/Platform.Ranges/RangeExtensions.h#L5-L23

do that like this:
```c++
template<typename T>
public: static T Difference(Range<T> range) { return range.Maximum - range.Minimum; }
```
Return type is `T`, but we can also use `auto`.

**Comment by uselessgoddess** [comment]
In the C++20 standard we can use 
```cpp
template<typename T> requires 
    requires(T maximum, T minimum) { maximum - minimum; }
public: static T Difference(Range<T> range) { return range.Maximum - range.Minimum; }
```

**Comment by Konard** [comment]
Yes, we can.
Looks like this issue was solved by #61
